MarsVars Tests

falconstryker opened the pull request:
Completed MarsVars unit tests. This includes a yml file for triggering github actions to test MarsVars too.

New variables were added to the dummy atmos_average and atmos_average_pstd files for these tests. I do not expect that to affect any other tests and based on my testing, it hasn't.

rurata commented:
MarsVars test failed on windows.
`STDERR: 
Changed to test directory: C:\Users\RUNNER~1\AppData\Local\Temp\MarsVars_test_12r2trqi
Running command: C:\hostedtoolcache\windows\Python\3.9.13\x64\python.exe D:\a\AmesCAP\AmesCAP\bin\MarsVars.py -h
Working directory: C:\Users\RUNNER~1\AppData\Local\Temp\MarsVars_test_12r2trqi
File exists check: True
STDOUT: 
STDERR: Traceback (most recent call last):
  File "D:\a\AmesCAP\AmesCAP\bin\MarsVars.py", line 494, in <module>
    args = parser.parse_args()
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\argparse.py", line 1825, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\argparse.py", line 1858, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\argparse.py", line 2067, in _parse_known_args
    start_index = consume_optional(start_index)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\argparse.py", line 2007, in consume_optional
    take_action(action, args, option_string)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\argparse.py", line 1935, in take_action
    action(self, namespace, argument_values, option_string)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\argparse.py", line 1099, in __call__
    parser.print_help()
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\argparse.py", line 2555, in print_help
    self._print_message(self.format_help(), file)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\argparse.py", line 2561, in _print_message
    file.write(message)
  File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u03bb' in position 3033: character maps to <undefined>`

falconstryker commented:
Fixed issues with MarsVars tests running on Windows. Issues were related to windows file paths, netCDF file locking, and some special characters. Fixed these in MarsVars.py. @rurata can squash and merge now.

Note the MarsFiles test just timed out. still working on improving that one.
